### PR TITLE
powsybl-tutorials: exec:java and readme

### PR DESCRIPTION
**Pull request by jonenst**
* **Please check if the PR fulfills these requirements** (please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)
- [x] The commit message follows our guidelines
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature + docs update

* **What is the current behavior?** (You can also link to an open issue here)
one must import the project in an IDE or know how to use maven to launch tutorials

* **What is the new behavior (if this is a feature change)?**
Allow to launch various tutorials with mvn exec:java, for example
```bash
# from the toplevel dir
$ mvn package exec:java -pl csv-exporter
```

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+## Environment requirements
+
+  * JDK *(1.8 or greater)*
+  * Maven *(3.3.9 or greater)*
+
+Most tutorials show simple code and can be run directly from maven using the exec:java goal:
+
+```
+$ mvn package exec:java -pl tutorial-folder
```
Comment (MioRtia) [inline README.md]: Make it explicit that `tutorial-folder` is a variable parameter:
> `$ mvn package exec:java -pl <TUTORIAL_FOLDER>`

File: README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+$ cat /tmp/test.csv 
+LineId,SubstationId1,SubstationId2,VoltageLevelId1,VoltageLevelId2,BusId1,BusId2,R,X,G1,B1,G2,B2
+NHV1_NHV2_1,P1,P2,VLHV1,VLHV2,NHV1,NHV2,3.00000,33.0000,0.00000,0.000193000,0.00000,0.000193000
+NHV1_NHV2_2,P1,P2,VLHV1,VLHV2,NHV1,NHV2,3.00000,33.0000,0.00000,0.000193000,0.00000,0.000193000
+```
+
+Alternatively, import the project in you favorite IDE and use its facilities to run code and start experimenting !
+
+Some tutorials (e.g. javafx-packager,  are not about running code but about packaging and therefore have different instructions. Please visit the https://www.powsybl.org/docs/tutorials page to find dedicated instructions.
```
Comment (MioRtia) [inline README.md]: typo: missing closing brackets:
> to find dedicated instructions).

File: README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+## Tutorials
+
+This document describes how to build and run small tutorial projects. For more in depth explanations on each tutorial, please visit the https://www.powsybl.org/docs/tutorials page.
+
+## Environment requirements
+
+  * JDK *(1.8 or greater)*
+  * Maven *(3.3.9 or greater)*
+
+Most tutorials show simple code and can be run directly from maven using the exec:java goal:
```
Comment (MioRtia) [inline README.md]: Make it explicit that this command must be done from the repository root (add ` cd powsybl-tutorials` before ?)

**Comment by jonenst** [comment]
Also maybe we can just advice to use mvn compile exec:java, this will make resources come from files instead of the jar I think, and it won't run the tests.

**Comment by jonenst** [comment]
> Also maybe we can just advice to use mvn compile exec:java, this will make resources come from files instead of the jar I think, and it won't run the tests.

The resources of the current project always come from the target/classes directory for maven-surefire-plugin and for maven-exec-plugin:java (it doesn't depend on the compile or package phase). (note: for integration tests (mvn verify) using maven-failsafe-plugin, they come from the jar, ie getResources returns a url like `jar:file:/tmp/foo/my-app/target/my-app-1.0-SNAPSHOT.jar!/toto.txt` )
they will come from the jar if one launches the jar manually, but no one does that.

Still it's better to run mvn compile exec:java to avoid confusing users with an extra jar and not run the tests.

**Comment by jonenst** [comment]
@MioRtia I addressed your remarks
When more tutorials are added, I think we should have an exhaustive list of the tutorials with a one line explanation + the link to the longer explanation. We need to first group all the ones launched with exec:java and then group the others. I think this should be in the toplevel readme.

**Comment by jonenst** [comment]
Note: for dependent projets in the reactor, the resources comes from the file if you didn't go to the package phase and from the jar if you did

```
$ mvn test exec:java
file:/tmp/foo/my-app/target/classes/toto.txt # from the surefire tests
file:/tmp/foo/my-app/target/classes/toto.txt # from the exec:java
```
```
$ mvn package exec:java
jar:file:/tmp/foo/my-app/target/my-app-1.0-SNAPSHOT.jar!/toto.txt # from the surefire tests
jar:file:/tmp/foo/my-app/target/my-app-1.0-SNAPSHOT.jar!/toto.txt # from the exec:java
```

**Pull request merged**